src/Orleans.Core/Serialization/SerializationManager.cs

```diff
@@ -222,6 +222,22 @@ public void Register(Type t, DeepCopier cop, Serializer ser, Deserializer deser)
             Register(t, cop, ser, deser, false);
         }
 
+        private object InitializeSerializer(Type serializerType)
+        {
+            if (this.ServiceProvider == null)
+                return null;
+
+            if (!serializerType.GetCustomAttributes(typeof(SerializerAttribute), true).Any())
+                return null;
+
+            var constructors = serializerType.GetConstructors();
+            if (constructors == null || constructors.Length < 1)
```

Comment:
would default constructors show up in this list even if they aren't explicitly defined in code?

Reply:
Yes. This will return an empty list only if the constructor is not accessible or the class is static.